### PR TITLE
Improve performance by not using cancellable manager for tap gesture

### PR DESCRIPTION
**Pull request by martingagnon**
## Description
Instead of adding the tapGesture in a cancellableManager, we remove it by class type.

## Motivation and Context
This PR https://github.com/mirego/trikot.streams/pull/98 is supposed to optimise labels (and other UI components) when there is no need of creating subscription (thus CancellableManager). However, for each UI components, we were creating a new CancellableManagerProvider that we were assigning in the `trikotInternalPublisherCancellableManager` (Created when needed). This means that each binding was creating 2 cancellableManager when they were not needed and at scale, it was impacting performance.

This reduce the pressure on the GC.

## How Has This Been Tested?
- [X] All features has been added/updated in sample application ( /sample )

Tested in Trikot.patron

## Types of changes
- [X] Bug fix (non-breaking change which fixes an issue)

### CODE REVIEW

**Comment by marcantoinefortier** [comment]
Could we just make a new `action` property that is not a Publisher, so that platforms can invoke this new action right away. 

For now we could juste make a default implementation in our ViewModels that does something like `Promise.from(this.action).onSuccess { action.invoke(context) }` so that we don't break the current behaviour?

I'm writing this without seeing the actual code, but does my "general idea" makes sense (knowing that we'd prefer the viewModel actions not to be observable)?

File: swift-extensions/UIViewExtensions.swift [hunk 1]
```diff
@@ -16,20 +17,13 @@ extension UIView {
...
             if !(self is UIControl) {
                 observe(viewModel.action) {[weak self] (value: ViewModelAction) in
                     guard let self = self else { return }
-                    let newCancellableManager = onTapResetableCancelableManager.cancelPreviousAndCreate()
-
+                    self.removePreviousRegisteredTapAction()
                     if value != ViewModelAction.Companion().None {
-                        let tapGestureReconizer = UITapGestureRecognizer(target: self, action: #selector(self.trikotOnViewTouchUp))
+                        let tapGestureReconizer = TrikotTapGestureRecognizer(target: self, action: #selector(self.trikotOnViewTouchUp))
                         self.addGestureRecognizer(tapGestureReconizer)
```
Comment (flambert) [inline swift-extensions/UIViewExtensions.swift:26]: Are you adding a TapGestureRecognizer for buttons too or is it only if it's a control without touch interactions?

Comment (martingagnon) [reply swift-extensions/UIViewExtensions.swift:26]: UIControls and Buttons uses native selector for actions. Its for views that does not have this.

**Comment by martingagnon** [comment]
> Could we just make a new `action` property that is not a Publisher, so that platforms can invoke this new action right away?
> 
> For now we could juste make a default implementation in our ViewModels that does something like `Promise.from(this.action).onSuccess { action.invoke(context) }` so that we don't break the current behaviour?
> 
> I'm writing this without seeing the actual code, but does my "general idea" makes sense (knowing that we'd prefer the viewModel actions not to be observable)?

What does it have to do with this PR?

Your Idea is however neat. However I see some pitfalls if this approach as it was the previous implementation.

```
val itemPublisher = fetchItem().startWith(NoItem)
...
val action = itemPublisher.map { if it == NoItem actionA else actionB }
```

The executed action will always be actionA.

**Comment by marcantoinefortier** [comment]
> What does it have to do with this PR?

I just feel like our current implementation is simply outdated and using a pattern that we don't like using anyway. I only took the oppurtinity to get my word out haha 😛 . 

I'll try something and open a PR with my proposition to address the "pitfall" you mentionned.

**Pull request merged**